### PR TITLE
fix: add support for multi-header with same key

### PR DESCRIPTION
**Pull request by rampagecong**
### Description

add support for multi-header with same key like `set-cookie`

Fixes # (issue)
https://github.com/apache/apisix-go-plugin-runner/issues/74
### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [x] I have explained the changes or the new features added to this PR
- [x] I have added tests corresponding to this change
- [x] I have updated the documentation to reflect this change
- [x] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

**Comment by rampagecong** [comment]
@spacewander Could you please take a look?

File: t/lib/ext-plugin.lua [hunk 1]
```diff
@@ -373,6 +373,40 @@ function _M.go(case)
                 {"X-Req", "bar"},
                 {"Content-Type", "application/json"},
                 {"Content-Encoding", "deflate"},
+                {"X-Same", "one"},
```
Comment (spacewander) [inline t/lib/ext-plugin.lua:376]: We don't need X-Same for other header tests?

Comment (rampagecong) [reply t/lib/ext-plugin.lua]: Sorry, I optimized the code.I divided it into three test cases based on this feature.
TEST 19: rewrite response header and call the upstream service
TEST 20: rewrite non-important response headers and call the upstream service
TEST 25: rewrite same response headers and call the upstream service

**Review by spacewander — APPROVED** [review]

File: t/lib/ext-plugin.lua (hunk 1 above)
Comment (spacewander) [reply t/lib/ext-plugin.lua]: Don't worry, it's just a small problem.

**Comment by spacewander** [comment]
@rampagecong 
Could you apply the same fix to other headers? Thanks!

**Pull request merged**